bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/community/CommunityMarketplaceAddonService.java

```diff
@@ -278,64 +280,69 @@ private String getContentType(@Nullable Integer category, List<String> tags) {
      * @param topic the topic
      * @return the list item
      */
-    private Addon convertTopicItemToAddon(DiscourseTopicItem topic, List<DiscourseUser> users) {
-        List<String> tags = Arrays.asList(Objects.requireNonNullElse(topic.tags, new String[0]));
-
-        String uid = ADDON_ID_PREFIX + topic.id.toString();
-        AddonType addonType = getAddonType(topic.categoryId, tags);
-        String type = (addonType != null) ? addonType.getId() : "";
-        String id = topic.id.toString(); // this will be replaced after installation by the correct id if available
-        String contentType = getContentType(topic.categoryId, tags);
-
-        String title = topic.title;
-        boolean compatible = true;
-
-        int compatibilityStart = topic.title.lastIndexOf("["); // version range always starts with [
-        if (topic.title.lastIndexOf(" ") < compatibilityStart) { // check includes [ not present
-            String potentialRange = topic.title.substring(compatibilityStart);
-            Matcher matcher = BundleVersion.RANGE_PATTERN.matcher(potentialRange);
-            if (matcher.matches()) {
-                try {
-                    compatible = coreVersion.inRange(potentialRange);
-                    title = topic.title.substring(0, compatibilityStart).trim();
-                    logger.debug("{} is {}compatible with core version {}", topic.title, compatible ? "" : "NOT ",
-                            coreVersion);
-                } catch (IllegalArgumentException e) {
-                    logger.debug("Failed to determine compatibility for addon {}: {}", topic.title, e.getMessage());
-                    compatible = true;
+    private @Nullable Addon convertTopicItemToAddon(DiscourseTopicItem topic, List<DiscourseUser> users) {
+        try {
+            List<String> tags = Arrays.asList(Objects.requireNonNullElse(topic.tags, new String[0]));
+
+            String uid = ADDON_ID_PREFIX + topic.id.toString();
+            AddonType addonType = getAddonType(topic.categoryId, tags);
+            String type = (addonType != null) ? addonType.getId() : "";
```

Comment:
```suggestion
            AddonType addonType = getAddonType(topic.categoryId, tags);
            if (addonType == null) {
                logger.debug("Ignoring topic '{}' because no add-on type could be found", topic.id);
                return;
            }
            String type = addonType.getId();
```

There is no need to continue processing if we know we'll fail later.